### PR TITLE
Tweak NanoUI Subsystem code

### PR DESCRIPTION
**Pull request by neersighted**

### CODE REVIEW

File: code/controllers/subsystem/nano.dm [hunk 1]
```diff
@@ -15,14 +15,12 @@ var/datum/subsystem/nano/SSnano
...
 
 /datum/subsystem/nano/fire() // Process UIs.
-	for(var/thing in SSnano.processing_uis)
-		if(thing)
-			var/datum/nanoui/ui = thing
-			if(ui.src_object && ui.user)
-				ui.process()
-				continue
-		processing_uis.Remove(thing)
+	for(var/datum/nanoui/ui in processing_uis)
```
Comment (duncathan) [inline code/controllers/subsystem/nano.dm]: This slows down the loop. The only thing you should be changing here is getting rid of the `if(thing)` on line 23.

Comment (neersighted) [inline code/controllers/subsystem/nano.dm]: How does this slow down the loop?

Comment (RemieRichards) [inline code/controllers/subsystem/nano.dm]: ~Byond~

Comment (Giacom) [inline code/controllers/subsystem/nano.dm]: !BYOND!

Comment (duncathan) [inline code/controllers/subsystem/nano.dm]: By doing `for(var/path/thing in list)` you're basically calling an istype() on every item in it. If you know that they're all a certain type, you don't need that check.

**Pull request merged**